devito/types/sparse.py

```diff
@@ -474,7 +474,7 @@ class SparseFunction(AbstractSparseFunction):
 
     def __init_finalize__(self, *args, **kwargs):
         super(SparseFunction, self).__init_finalize__(*args, **kwargs)
-
+        self.interpolator = LinearInterpolator(self.grid, self)
```

Comment:
I would add a class attribute for this

```
class SparseFunction(...):

     _default_interpolator = LinearInterpolator

     def __init_finalize__(...):
           ...
           self.interpolator = self._default_interpolator(...)
           # or self.interpolator = kwargs.get('interpolator', self._default_interpolator)()
```

for subclasses and user it will then be possible to specialize what interpolator to use, as long as the interpolator interface is honored (that's why I was suggesting the common abstract class a few comments ago)

Reply:
Can be kept as a TODO because the constructors for LinearInterpolator and PrecomputedInterpolator take different arguments for now. Not easy to homogenise yet.

Reply:
See #1046